include/ioh/problem/pbo/N_queens.hpp

```diff
@@ -0,0 +1,124 @@
+#pragma once
+#include "pbo_problem.hpp"
+
+
+namespace ioh
+{
+    namespace problem
+    {
+        namespace pbo
+        {
+            class NQueens : public PBOProblem<NQueens>
+            {
+            protected:
+
+                // TODO: why not use std::max?
+                static double _max(double element1, double element2)
+                {
+                    if (element1 > element2)
+                    {
+                        return element1;
+                    }
+                    return element2;
+                }
+
+                std::vector<double> evaluate(const std::vector<int> &x) override
+                {
+                    int j, i;
+                    auto n_ = x.size();
+                    const auto N_queens = static_cast<int>(sqrt(static_cast<double>(n_)) + 0.5);
+                    auto number_of_queens_on_board = 0;
+                    auto k_penalty = 0.0;
+                    auto l_penalty = 0.0;
+                    auto raws_penalty = 0.0;
+                    auto columns_penalty = 0.0;
+                    auto indx = 0;
+                    const auto C = static_cast<float>(N_queens);
+
+                    if (floor(sqrt(static_cast<double>(n_))) != sqrt(static_cast<double>(n_)))
+                    {
+                        common::log::error("Number of parameters in the N Queen problem must be a square number");
+                    }
+
+                    for (auto index = 0; index < n_; index++)
```

Comment:
You can write these statements more consisly with std::accumulate:   int number_of_queens_on_board = std::accumulate(x.begin(), x.end(), 0);